waku/factory/node_factory.nim

```diff
@@ -113,6 +113,12 @@ proc initNode(
 
 ## Mount protocols
 
+proc getNetworkShards*(conf: WakuNodeConf): uint32 =
+  if conf.networkShards != 0:
+    return conf.networkShards
+  # If conf.networkShards is not set, use the number of shards configured as networkShards
+  return uint32(max(conf.shards) + 1)
```

Comment:
I'm not sure that would be intuitive for the user. I would suggest requiring this value if using custom network config.

Reply:
We currently use the length of pubsub topics configured. I changed this pattern because if we configure shards [0, 3] and we have 2 network shards it wouldn't make sense, so I went for the max + 1.

I agree it might not be intuitive, but I was afraid of requiring that parameter and making it a breaking change. If we require it, all the tests and uses in shards different than 1 would be broken by this new requirement.

@alrevuelta what's your take on this?

Maybe a good alternative would be to log a warning when we set that number for them if they didn't

Reply:
I guess I don't like that kind of magic. In this case it's fine the stakes are low anyway.

Reply:
I agree 100% that it's better to avoid magic. But as you said, stakes are low and it doesn't break things :)